snaps/how-to/integrate-custom-evm-accounts.md

```diff
@@ -0,0 +1,206 @@
+
+## Terminology
+
+Let's introduce some terminology used across the Keyring API:
+
+- **Blockchain account**: An object in a single blockchain, representing an account, with its balance, nonce, etc.
+- **Request**: A request sent by a Dapp to MetaMask.
+- **Keyring account**: An account model that represents one or more blockchain accounts.
+- **Keyring snap**: A snap that implements the Keyring API.
+- **Keyring request**: A request from MetaMask to a keyring snap, to perform an action on or using a keyring account. It wraps the original request sent by the Dapp and adds some metadata to it.
+
+## Components diagram
+
+When interacting with accounts managed by a Keyring snap, we will encounter the following components:
+
+![Keyring snap component diagram](../assets/keyring/components-diagram.png)
+
+- **User**: The web3 user interacting with the snap, the Dapp, and MetaMask.
+- **Dapp**: The web3 application that is requesting an action to be performed on an account.
+- **MetaMask**: The web3 provider that dApps connect to. It routes requests to the keyring snaps and lets the user perform some level of account management.
+- **Snap**: A snap that implements the Keyring API to manage the user's accounts, and to handle requests that use these accounts.
+- **Snap UI**: The snap's UI component that allows the user to interact with the snap to perform custom operations on accounts and requests.
+
+## Implementing a `Keyring` class
+
+The first step for creating a Keyring snap is to implement the [`Keyring` interface](../reference/keyring-api/modules.md#keyring). This interface describes all the methods necessary to make your custom EVM accounts work inside MetaMask with your own logic. The next sections will go over the methods of the `Keyring` interface by describing the different flows that it handles.
+
+## Snap account creation flow
+
+The first interaction between users and your Keyring snap will be the snap account creation process. Here is a diagram of the process:
+
+![Keyring snap account creation flow](../assets/keyring/create-account-flow.png)
+
+The MetaMask account selection modal has an option called "Add snap account":
+
+![add snap account option](../assets/keyring/add-snap-account.png)
+
+This option will show a list of keyring snaps, but will ultimately redirect the user to the companion Dapp for your Keyring snap. That is, a Dapp that contains all the UI to configure and manage the Keyring snap.
+
+On that Dapp, you'll present a custom user interface allowing the user configure their custom EVM account. The Dapp will make use of the [`KeyringSnapRpcClient`'s `createAccount` method](../reference/keyring-api/classes/KeyringSnapRpcClient.md#createaccount), which will call your `Keyring` class' method of the same name. An example of this can be found in the [Simple keyring snap companion Dapp](https://github.com/MetaMask/snap-simple-keyring/blob/d3f7f0156c59059c995fea87f90a3d0ad3a4c135/packages/site/src/pages/index.tsx#L136).
```

Comment:
Getting a 404 @ https://github.com/MetaMask/metamask-docs/blob/zs/keyring-doc/snaps/reference/keyring-api/classes/KeyringSnapRpcClient.md#createaccount as the `keyring-api` folder does not seem to exist.

Reply:
That's normal, as it gets created at build time from a Git submodule!